common/authorization/defaultClaimMapper.go

```diff
@@ -0,0 +1,121 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package authorization
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+// Default claim mapper that gives system level admin permission to everybody
+type defaultClaimMapper struct {
+	keyProvider TokenKeyProvider
+}
+
+func NewDefaultClaimMapper(provider TokenKeyProvider) ClaimMapper {
+	return &defaultClaimMapper{keyProvider: provider}
+}
+
+var _ ClaimMapper = (*defaultClaimMapper)(nil)
+
+func (a *defaultClaimMapper) GetClaims(authInfo AuthInfo) (*Claims, error) {
+
+	claims := Claims{}
+
+	if authInfo.authToken != "" {
```

Comment:
maybe I missed it completely...where is tlsSubject used?

Reply:
It is not used in the default implementations of claim mapper and authorizer, at least not yet. But it would be used for mTLS-based authorization that some customers employ today.